components/event-gateway/internal/externalapi/errorhandler_test.go

```diff
@@ -0,0 +1,43 @@
+package externalapi
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/kyma-project/kyma/components/event-gateway/internal/httperrors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestErrorHandler_ServeHTTP(t *testing.T) {
+	t.Run("Should always respond with given error and status code", func(t *testing.T) {
+
+		r := mux.NewRouter()
+
+		r.NotFoundHandler = NewErrorHandler(404, "Requested resource could not be found.")
+		ts := httptest.NewServer(r)
+		defer ts.Close()
+
+		// when
+		res, err := http.Get(ts.URL + "/wrong/path")
+
```

Comment:
"err" could be != nil here

Reply:
This is a code copy from the `gateway` component.
However, this is fixed because it is minor.